qurami/strings2pot
# OO version

This branch refactors the strings2pot package using OO code, a modular approach, tests.

This change is necessary as pre-requirement to create new processors.

To test the version:
- Download the code
- Enter the directory
- Write a `test.strings` test source file such as:

```
/* Insert Element menu item */

"Insert Element" = "Insert Element";

/* Test string with a placeholder */

"Test string with a %@ here" = "Test string with a %@ here";
```
- Launch the processor via the following command:

`python strings2pot.py test.strings ios.pot`

- Check that the `ios.pot` file exists and it is a valid POT file


Same should be done using a `test.xml` file such as:

```xml
<?xml version="1.0" encoding="utf-8"?>
<resources>
  <string name="string_name">Hello %1$s, nice to see you here!</string>
</resources>
```

And then running the processor via 

`python strings2pot.py test.xml android.pot`.

Run tests via the `make test` command.

No external packages are required for this script to run.
